core/kernel/tee_ta_manager.c

```diff
@@ -738,6 +737,16 @@ TEE_Result tee_ta_open_session(TEE_ErrorOrigin *err,
 			res = TEE_ERROR_TARGET_DEAD;
 		}
 
+#if defined(CFG_FTRACE_SUPPORT) && defined(CFG_FTRACE_DUMP_EVERY_ENTRY)
+		if (!panicked) {
+			if (ts_ctx && ts_ctx->ops->dump_ftrace) {
+				ts_push_current_session(&s->ts_sess);
+				ts_ctx->ops->dump_ftrace(s->ts_sess.ctx);
+				ts_pop_current_session();
+			}
+		}
+#endif
+
```

Comment:
This pattern is used 3 times. Can you please make it a static function?
```C
static void dump_ftrace(struct tee_ta_session *s __maybe_unused)
{
#if defined(CFG_FTRACE_SUPPORT)
	struct ts_ctx *ts_ctx = s->ts_sess.ctx;

	if (ts_ctx && ts_ctx->ops->dump_ftrace) {
		ts_push_current_session(&s->ts_sess);
		ts_ctx->ops->dump_ftrace(ts_ctx);
		ts_pop_current_session();
	}
#endif
}
```

Then:

```C
		if (!panicked && IS_ENABLED(CFG_FTRACE_DUMP_EVERY_ENTRY))
			dump_ftrace();
```